sdk/java/pulumi/src/main/java/com/pulumi/deployment/internal/DeploymentImpl.java

```diff
@@ -164,6 +165,14 @@ public static DeploymentImpl fromEnvironment() {
         return impl;
     }
 
+    @InternalUse
+    public static DeploymentImpl fromInline(InlineDeploymentSettings settings) {
+        var state = DeploymentState.fromInline(settings);
+        var impl = new DeploymentImpl(state);
+        DeploymentInstanceHolder.setInstance(new DeploymentInstanceInternal(impl));
```

Comment:
Have you tried multiple calls to `Construct` within the same process (e.g. creating two separate instances of an MLC from the Pulumi program)?

I am hitting the following when I have multiple operations that involve the inline program (e.g. a preview and up) in the same process:

https://github.com/pulumi/pulumi-java/blob/85e505ddcadbd6c674b36c61af1dc08ad5e49b69/sdk/java/pulumi/src/main/java/com/pulumi/deployment/internal/DeploymentInstanceHolder.java#L39

I haven't dug in to see if/how .NET avoids it.

Reply:
I confirm I can see the same, thank you. .NET seems to work fine - I'll dig further.